tensorly/decomposition/_parafac2.py

```diff
@@ -45,8 +45,9 @@ def _update_imputed(tensor_slices, mask, decomposition, method):
     if method == "mode-2":
 
         for slice_no, (slice, slice_mask) in enumerate(zip(tensor_slices, mask)):
+
             slice_mean = tl.sum(slice * slice_mask) / (
-                tl.prod(slice.shape) - len(tl.where(slice_mask == 0)[0])
+                int(T.prod(T.tensor(slice.shape))) - len(T.where(slice_mask == 0)[0]) # tensorflow required this casting to int
             )
```

Comment:
Since the mask is 1 for all observed values and 0 for all non-observed values, we could probably simplify this a bit. Maybe this will work as a shorter alternative?
```suggestion
            slice_mean = tl.sum(slice * slice_mask) / int(T.sum(slice_mask))  # Cast to int due to TensorFlow
```

Reply:
Addressed that one as well :) Good eye!